manifests/init.pp

```diff
@@ -77,11 +81,15 @@
   Optional[Integer[1]]               $max_memory                 = undef,
   Optional[Integer[1]]               $max_files                  = undef,
   Stdlib::Absolutepath               $distfiles_cache            = '/usr/ports/distfiles',
+  Optional[String[1]]                $git_baseurl                = undef,
+  Optional[String[1]]                $git_portsurl               = undef,
```

Comment:
can you use Stdlib::Httpsurl?

Reply:
I am not sure about this one: while HTTPS is probably the most commonly used protocol, some users/organizations have a clone of the FreeBSD ports mirror on GitHub to add local changes and might prefer to clone using deploy keys over SSH.  By not validating the format here, the user can use anything supported by git current and future (i.e. one might still use git:// URI).